### PR TITLE
Add sof-v2.9 for Intel hardware

### PR DESCRIPTION
**Pull request by kv2019i**

### CODE REVIEW

**Comment by kv2019i** [comment]
V2 uploaded, now tests pass.

**Comment by kv2019i** [comment]
V3: fix tarball_multi_releases
  - but BATS tests are failing

V4: fix BATS tests again

**Comment by kv2019i** [comment]
V5: one more fixup to the sof-ace-tplg symlink handling

**Review by ujfalusi — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Comment by kv2019i** [comment]
V6: 
- updated the communiy-signed binaries (now binary uses "v2.9" tag instead of "v2.9-rc1" so there's more obvious match of binaries... both tags point to same git commit so the compiled code is identical, but as the literals are of different size, running a binary diff will show lot of diffences -> now fixed)

File: tarball_multi_releases.bash [hunk 1]
```diff
@@ -125,6 +125,15 @@ main()
     check_symlinks "$archive_name" ||
         die "Found some broken symbolic links after combining\n"
 
+    # 2024.03 compatibility hack, add a symlink for sof-ace-tplg as per
+    # the compatibility note for Intel MTL in
+    # https://thesofproject.github.io/latest/getting_started/intel_debug/introduction.html#user-space-and-filesystem-requirements
+    ( cd "${archive_name}"
+      if test -e "sof-ipc4-tplg-WIP" -a ! -e "sof-ace-tplg-WIP" ; then
```
Comment (marc-hb) [inline tarball_multi_releases.bash:132]: shellcheck would likely say:

```suggestion
      if test -e "sof-ipc4-tplg-WIP" && ! test -e "sof-ace-tplg-WIP" ; then
```

File: tests/tests.bats [hunk 2]
```diff
@@ -198,7 +198,9 @@ test_install_one_version()
 
     tar xf sof-bin-"$ver".tar.gz
 
-    diff -qr "$EXTR_REFS"/sof-bin-"$ver"  "$(pwd)/sof-bin-$ver"/
+    # install.sh was updated in 2024.03, so it won't match
+    # anymore
+    diff -qr --exclude="install.sh" "$EXTR_REFS"/sof-bin-"$ver"  "$(pwd)/sof-bin-$ver"/
```
Comment (marc-hb) [inline tests/tests.bats:203]: Can you please try this instead?
```shell
git show v2023.12.1:install.sh > "$(pwd)/sof-bin-$ver"/install.sh
```

File: v2.9.x/tools-v2.9/mtrace-reader.py
Comment (marc-hb) [inline v2.9.x/tools-v2.9/mtrace-reader.py:1]: Was there any tools change between v2.8 and v2.9? If not, can you just re-use `tools-v2.8` and reduce noice?

**Review by marc-hb — COMMENTED** [review]
> > updated the communiy-signed binaries (now binary uses "v2.9" tag instead of "v2.9-rc1" so there's more obvious match of binaries...

Why keep the v2.9-rc1 binaries ?

File: v2.9.x/tools-v2.9/mtrace-reader.py
Comment (kv2019i) [reply v2.9.x/tools-v2.9/mtrace-reader.py:1]: @marc-hb There was a minor change in sof-ctl, so we need to update.

File: tarball_multi_releases.bash (hunk 1 above)
Comment (kv2019i) [reply tarball_multi_releases.bash:132]: @marc-hb Hmm, shellcheck seems happy with original.

**Comment by kv2019i** [comment]
V7:
- addressed Marc's comment in test code
- remove the rc1 community binaries from git history
- no changes to FW binaries or topologies

@marc-hb please check

**Review by marc-hb — APPROVED** [review]

**Pull request merged**